### PR TITLE
Upload-Flow installation bug fix

### PR DESCRIPTION
**Pull request by guyfreund**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
When trying to install **all packs** the current script is iterating over the content repo and checking the `serverMinVersion` for each pack. This field is usually populated in the upload-flow (and minorly exists in the repo) we end up installing packs with fromversion > 6.0 on a 6.0 server. This is causing the installation to fail.

## Description
Now iterating over the build bucket that was configured in a pre-job to the installation job.

## Circle Successful Run
https://app.circleci.com/pipelines/github/demisto/content/99975/workflows/b6f99c3b-3f36-4027-a3b1-bfa6e8a6f447

## GitLab Successful Run
https://code.pan.run/xsoar/content/-/pipelines/1371791

### CODE REVIEW

**Comment by guyfreund** [comment]
PR is ready to be reviewed

File: Tests/Marketplace/marketplace_constants.py [hunk 1]
```diff
@@ -54,6 +54,7 @@ class GCPConfig(object):
     BASE_PACK = "Base"  # base pack name
     INDEX_NAME = "index"  # main index folder name
     CORE_PACK_FILE_NAME = "corepacks.json"  # core packs file name
+    BUILD_BUCKET_PACKS_ROOT_PATH = 'content/builds/{}/{}/content/packs'
```
Comment (yaakovi) [inline Tests/Marketplace/marketplace_constants.py:57]: please use keywords in the format so the constant's usage will be nicer to read
 
`BUILD_BUCKET_PACKS_ROOT_PATH.format(branch=branch_name, build=build_num`

```suggestion
    BUILD_BUCKET_PACKS_ROOT_PATH = 'content/builds/{branch}/{build}/content/packs'
```

File: Tests/configure_and_test_integration_instances.py [hunk 2]
```diff
@@ -983,9 +983,12 @@ def get_pack_ids_to_install():
         #  END CHANGE ON LOCAL RUN  #
 
 
-def nightly_install_packs(build, install_method=install_all_content_packs, pack_path=None, service_account=None):
+def nightly_install_packs(build, install_method=None, pack_path=None, service_account=None):
```
Comment (yaakovi) [inline Tests/configure_and_test_integration_instances.py:986]: why did you remove the default?
this basically means you can change it to a positional argument now

File: Tests/Marketplace/configure_and_install_packs.py [hunk 3]
```diff
@@ -5,20 +5,31 @@
...
                                           ' AMI environment.', default="Server Master")
     parser.add_argument('-s', '--secret', help='Path to secret conf file')
     parser.add_argument('--branch', help='GitHub branch name', required=True)
     parser.add_argument('--build_number', help='CI job number where the instances were created', required=True)
+    parser.add_argument('--service_account', help="Path to gcloud service account, is for circleCI usage. For local "
+                                                  "development use your personal account and authenticate using Google "
+                                                  "Cloud SDK by running: `gcloud auth application-default login` and "
+                                                  "leave this parameter blank. For more information go to: "
+                                                  "https://googleapis.dev/python/google-api-core/latest/auth.html",
+                        required=True)
```
Comment (yaakovi) [inline Tests/Marketplace/configure_and_install_packs.py:27]: `leave this parameter blank` and `required=True`?

File: Tests/Marketplace/search_and_install_packs.py [hunk 4]
```diff
@@ -480,32 +481,46 @@ def install_all_content_packs_for_nightly(client: demisto_client, host: str, ser
...
-                pack_version = pack_metadata.get('currentVersion')
-                server_min_version = pack_metadata.get('serverMinVersion', '6.0.0')
-                hidden = pack_metadata.get('hidden', False)
+    storage_client = init_storage_client(service_account)
+    build_bucket = storage_client.bucket(GCPConfig.CI_BUILD_BUCKET)
+    GCPConfig.STORAGE_BASE_PATH = bucket_packs_root_path  # Setting Storage base path
+    index_folder_path, _, _ = download_and_extract_index(build_bucket, extract_destination_path)
+
+    for pack_id in os.listdir(index_folder_path):
+        if os.path.isdir(os.path.join(index_folder_path, pack_id)):
```
Comment (yaakovi) [inline Tests/Marketplace/search_and_install_packs.py:507]: shouldn't you keep the `IGNORED_FILES` check?

Comment (guyfreund) [reply Tests/Marketplace/search_and_install_packs.py:507]: As we are not iterating over the repo, but over the index, we don't need this check. See https://github.com/demisto/content/blob/master/Tests/Marketplace/upload_packs.py#L45

File: Tests/configure_and_test_integration_instances.py (hunk 2 above)
Comment (guyfreund) [reply Tests/configure_and_test_integration_instances.py:986]: Yes, this default value is out of use during all calls to this function

**Review by ShahafBenYakir — APPROVED** [review]
